scripts/create-references.js

```diff
@@ -11,55 +10,42 @@ const config = JSON.parse(fs.readFileSync('tsconfig.json').toString());
 config.files = [];
 config.references = [];
 
-(async function() {
-
-    const { stdout, stderr } = await exec('yarn workspaces info --json');
-    
-    const lines = stdout.split('\n');
-    const depthTree = lines.slice(1, lines.length - 2).join('\n');
-    let workspaces = undefined;
-
-    // sometimes yarn prints this line: yarn run v1.22.10 and sometimes not
-    try {
-        workspaces = JSON.parse(depthTree);
-    } catch (e) {
-        workspaces = JSON.parse(stdout);
-    }
-
-    for (const name in workspaces) {
-        const workspace = workspaces[name];
-        const location = path.resolve(process.cwd(), workspace.location);
-        const tsconfigPath = path.resolve(location, 'tsconfig.json');
-
-        if (fs.existsSync(tsconfigPath)) {
-            config.references.push({
-                path: workspace.location,
-            });
-            const workspaceConfig = JSON.parse(
-                fs.readFileSync(tsconfigPath).toString(),
-            );
-            workspaceConfig.compilerOptions.composite = true;
-            workspaceConfig.references = [];
-            for (const dependency of workspace.workspaceDependencies) {
-                const dependecyLocation = path.resolve(
-                    process.cwd(),
-                    workspaces[dependency].location,
-                );
-                if (
-                    fs.existsSync(
-                        path.resolve(dependecyLocation, 'tsconfig.json'),
-                    )
-                ) {
-                    workspaceConfig.references.push({
-                        path: path.relative(location, dependecyLocation),
-                    });
-                }
-            }
-            fs.writeFileSync(
-                tsconfigPath,
-                JSON.stringify(workspaceConfig, undefined, 4),
-            );
+(async function () {
+  const { stdout, stderr } = await exec('yarn workspaces info --json');
+
+  const lines = stdout.split('\n');
```

Comment:
Apparently, '\n' line separator does not work on all devices. Although I highly doubt an Atari TOS or MS DOS machine will ever visit this site, an older version of MacOS might and that only recognizes '\r' as end-of-line char.

For accessibility sake, to support all possible end-of-line characters ever, the value here should be `/\r\n|\n\r|\n|\r/g` (note, the order of the chars in this string is very important, I recommend copy-pasting it directly).

Here is the [source](https://dirask.com/posts/JavaScript-how-to-split-string-by-newline-k1wEQp).

Reply:
It adds a `\n` and fs will make sure this `\n` is properly written to the file.

Reply:
I see that, and the `.join('\n');` function below will work as expected as it is still a valid character. However I'm not sure whether the `.split('\n');` function will always work because of the different linebreaks in some OS.

It is a niche to take this into account, but since this is a gov site I think we'd want to take into account as many possible OSes, especially as this is a dev script and some devs are infamous for working on some arbitrary OS (right, @bddjong :)).

Reply:
I've updated the split with the characters you mentioned